---
bool value
string message
bool success